### ISSUE TITLE
Non-validating peer shows many "not enough seals to seal block" messages during sync

### ISSUE THREAD

**Issue by dankostiuk** [0xPolygon/polygon-edge#986]
# Non-validating peer shows many "not enough seals to seal block" messages during sync
## Description

After migrating to v0.6.0, we've observed  there to be many of the following WARN messages:
```
Nov 30 23:23:58 ip-172-31-41-130 main[758992]: 2022-11-30T23:23:58.963Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAmJUVAsjJwd6ah9JC1xHtwKREwzQHmQMLpoi6pdd4pf5bF="unable to verify block, failed to verify the header: not enough seals to seal block"
```

## Your environment
- OS and version. `Ubuntu 20`
- The version of the Polygon Edge.   `v0.6.0`
- The branch that causes this issue. `develop`
- Locally or Cloud hosted (which provider). `AWS ec2`

## Steps to reproduce

1. Spin up chain with 6 validators and 1 non-validator peer
2. Observe that the peer cannot sync to the current block, and we see many instances of WARN messages:
```
unable to verify block, failed to verify the header: not enough seals to seal block"
```

## Expected behavior

I expect non-validating peers to behave as they did prior to v0.6.0, syncing to the latest block without these WARN messages.

**Comment by ivanbozic21** [comment]
Hi @dankostiuk 

Can you confirm from which version and branch did you upgrade from?
Also, can you provide me with the debug logs as well in order to check them completely.

**Comment by dankostiuk** [comment]
We are running off a branch off the commit hash below:

```
[VERSION INFO]
Release version = v0.6.0
Git branch      = HEAD
Commit hash     = e25825cb96ff35db308af06deee69acd5caea9b7
```

as well as these cherry-picked changes by Kourin in hopes to solve our `consensus/metadata` issue: https://github.com/0xPolygon/polygon-edge/pull/839 

As we recently upgraded from v0.5.1 to v0.6.0, we have a few 3rd party peers still running v0.5.1 - which is probably why we are seeing so many `received status from non-connected peer` debug statements.

Upon further inspection, this non-validator node is no longer writing new blocks either as the following curl command keeps return the last block that was written (0xa05458):
```
curl --location --request POST 'localhost:10002' \
--header 'Content-Type: application/json' \
--data-raw '{"id":1,"jsonrpc":"2.0","method":"eth_blockNumber","params":[]}'
```

Additional logging:
```
Dec 01 15:21:38 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:38.606Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAm4cGCPV28nqTYgyPinsWUZGsze1eYJezNRf5P1RjBBRXm="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.240Z [DEBUG] polygon.network.identity: Conn: peer=16Uiu2HAmNbmbqa6LsUter8Ex7ybuzJWVRTJ5xCGtnVnww7gi2FEQ direction=Inbound
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.268Z [DEBUG] polygon.network: open stream: protocol=/id/0.1 peer=16Uiu2HAmNbmbqa6LsUter8Ex7ybuzJWVRTJ5xCGtnVnww7gi2FEQ
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.296Z [INFO]  polygon.network: Peer connected: id=16Uiu2HAmNbmbqa6LsUter8Ex7ybuzJWVRTJ5xCGtnVnww7gi2FEQ
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.296Z [DEBUG] polygon.network: Dialing peer [{16Uiu2HAmNbmbqa6LsUter8Ex7ybuzJWVRTJ5xCGtnVnww7gi2FEQ: [/ip4/54.145.42.157/tcp/10001]}] as local [16Uiu2HAmSYcSPH2ecd9cUUXPdyCUQDEzjJX1kts7NdeLaFpNsfrU]
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.309Z [WARN]  polygon.syncer: failed to get peer status, skip: id=16Uiu2HAmNbmbqa6LsUter8Ex7ybuzJWVRTJ5xCGtnVnww7gi2FEQ err="failed to open a stream, err protocol not supported"
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.855Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAmJUVAsjJwd6ah9JC1xHtwKREwzQHmQMLpoi6pdd4pf5bF="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.858Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAm6PeQd5hkLH7CvasuzJWnVgPcmzjZD5Q29bKb1xWXL3nh="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.859Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.860Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmA7t9aGbpzsabd7SdmDTh1oNo9Hayd11FtKEVPM5Jd8T4
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.861Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm7AkeKv4TPXPoxXrR8SUvoTSpHQQd4z9texdHp8EQ4oXv
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.862Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.862Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAkydREVfmMi7x4HJvyrykoqj9dDyPBsvroTNx3BnmMfnez
Dec 01 15:21:39 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:39.901Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmQ2rx4G6ZyYteotMmQiVVawvRpZmYVasBePP3gS5UZHeq
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.876Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAmRYg8TwmgFRcP2a5f3FskBKZwD3j2wq3iWtJh5gwdcYuv="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.877Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmA7t9aGbpzsabd7SdmDTh1oNo9Hayd11FtKEVPM5Jd8T4
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.877Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.878Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm7AkeKv4TPXPoxXrR8SUvoTSpHQQd4z9texdHp8EQ4oXv
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.879Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAkydREVfmMi7x4HJvyrykoqj9dDyPBsvroTNx3BnmMfnez
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.879Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.886Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAmAYbpcB42mAbMtQJKjeXYWqHqvS8tSm485KMDjazHQWdE="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:41 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:41.939Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmQ2rx4G6ZyYteotMmQiVVawvRpZmYVasBePP3gS5UZHeq
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.303Z [DEBUG] polygon.network.discovery: Querying a peer for near peers: peer=16Uiu2HAmAYbpcB42mAbMtQJKjeXYWqHqvS8tSm485KMDjazHQWdE
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.305Z [DEBUG] polygon.network.discovery: Found new near peers: peer=15
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.305Z [DEBUG] polygon.network: Dialing peer [{16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH: [/ip4/3.99.223.142/tcp/1478]}] as local [16Uiu2HAmSYcSPH2ecd9cUUXPdyCUQDEzjJX1kts7NdeLaFpNsfrU]
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.889Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAm81sJYmkYCEu13uYJxP8bMCKQ8Xks7eNKxskBjm5muWYz="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.891Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm7AkeKv4TPXPoxXrR8SUvoTSpHQQd4z9texdHp8EQ4oXv
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.892Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmA7t9aGbpzsabd7SdmDTh1oNo9Hayd11FtKEVPM5Jd8T4
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.892Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.893Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.893Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAkydREVfmMi7x4HJvyrykoqj9dDyPBsvroTNx3BnmMfnez
Dec 01 15:21:43 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:43.935Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmQ2rx4G6ZyYteotMmQiVVawvRpZmYVasBePP3gS5UZHeq
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.904Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAm81sJYmkYCEu13uYJxP8bMCKQ8Xks7eNKxskBjm5muWYz="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.905Z [ERROR] polygon.sync-peer-client: failed to get block from gRPC stream: peer=16Uiu2HAm81sJYmkYCEu13uYJxP8bMCKQ8Xks7eNKxskBjm5muWYz err="rpc error: code = Canceled desc = grpc: the client connection is closing"
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.907Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmA7t9aGbpzsabd7SdmDTh1oNo9Hayd11FtKEVPM5Jd8T4
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.908Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm7AkeKv4TPXPoxXrR8SUvoTSpHQQd4z9texdHp8EQ4oXv
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.908Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAkydREVfmMi7x4HJvyrykoqj9dDyPBsvroTNx3BnmMfnez
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.909Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.916Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:45 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:45.947Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmQ2rx4G6ZyYteotMmQiVVawvRpZmYVasBePP3gS5UZHeq
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.919Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAmRYg8TwmgFRcP2a5f3FskBKZwD3j2wq3iWtJh5gwdcYuv="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.922Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAm4cGCPV28nqTYgyPinsWUZGsze1eYJezNRf5P1RjBBRXm="unable to verify block, failed to verify the header: not enough seals to seal block"
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.923Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.924Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAm7AkeKv4TPXPoxXrR8SUvoTSpHQQd4z9texdHp8EQ4oXv
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.924Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmA7t9aGbpzsabd7SdmDTh1oNo9Hayd11FtKEVPM5Jd8T4
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.925Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.925Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAkydREVfmMi7x4HJvyrykoqj9dDyPBsvroTNx3BnmMfnez
Dec 01 15:21:47 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:47.995Z [DEBUG] polygon.sync-peer-client: received status from non-connected peer, ignore: id=16Uiu2HAmQ2rx4G6ZyYteotMmQiVVawvRpZmYVasBePP3gS5UZHeq
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.304Z [DEBUG] polygon.network.discovery: Querying a peer for near peers: peer=16Uiu2HAm6PeQd5hkLH7CvasuzJWnVgPcmzjZD5Q29bKb1xWXL3nh
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.305Z [DEBUG] polygon.network.discovery: Found new near peers: peer=15
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.306Z [DEBUG] polygon.network: failed to dial: addr="{16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH: [/ip4/3.99.223.142/tcp/1478]}"
Dec 01 15:21:48 ip-172-31-41-130 main[858691]:   err=
Dec 01 15:21:48 ip-172-31-41-130 main[858691]:   | failed to dial 16Uiu2HAmMwgnbYHEVBD98vQsmkZahkeHunBj7fPgq6vGKzxBY8xH:
Dec 01 15:21:48 ip-172-31-41-130 main[858691]:   |   * [/ip4/3.99.223.142/tcp/1478] dial tcp4 0.0.0.0:10001->3.99.223.142:1478: i/o timeout
Dec 01 15:21:48 ip-172-31-41-130 main[858691]:
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.306Z [DEBUG] polygon.network: Dialing peer [{16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf: [/ip4/3.99.20.25/tcp/10001]}] as local [16Uiu2HAmSYcSPH2ecd9cUUXPdyCUQDEzjJX1kts7NdeLaFpNsfrU]
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.320Z [DEBUG] polygon.network.identity: Conn: peer=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf direction=Outbound
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.326Z [DEBUG] polygon.network: Dialing peer [{16Uiu2HAkvJBGjsPfyJZMuBQCHohwsby75jiRqd1r1UVHBxitL77S: [/ip4/44.193.217.162/tcp/10001]}] as local [16Uiu2HAmSYcSPH2ecd9cUUXPdyCUQDEzjJX1kts7NdeLaFpNsfrU]
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.326Z [DEBUG] polygon.network: open stream: protocol=/id/0.1 peer=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.330Z [INFO]  polygon.network: Peer connected: id=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.331Z [DEBUG] polygon.network: open stream: protocol=/syncer/0.2 peer=16Uiu2HAm9HKUmFCACg5fnDQqfwcMUasV5yUsjFZSnHQiUEKC4Puf
Dec 01 15:21:48 ip-172-31-41-130 main[858691]: 2022-12-01T15:21:48.334Z [WARN]  polygon.syncer: failed to complete bulk sync with peer, try to next one: peer ID=error 16Uiu2HAm6PeQd5hkLH7CvasuzJWnVgPcmzjZD5Q29bKb1xWXL3nh="unable to verify block, failed to verify the header: not enough seals to seal block"
```

**Comment by laviniat1996** [comment]
Hey @dankostiuk did you switch to BLS after upgrading to v0.6.0?

**Comment by dankostiuk** [comment]
We kept the genesis.json the same, so using ECDSA as I believe that is the default.

Update - I took a backup of a validator's data-dir and restored directly onto the non-validating peer and the issue seemed to go away. I am no longer seeing the `not enough seals to seal block` either. It seems the non-validating peer's snapshot assumed there to be 9 validators total, so it's own snapshot was not updated for some reason. For now things to be working fine though so feel free to close this. Thanks @laviniat1996 .
